src/loader.rs

```diff
@@ -176,55 +176,104 @@ impl<'input> Loader<'input> {
                         None
                     };
                 }
-                YamlEvent::DocumentStart => continue,
+                YamlEvent::DocumentStart | YamlEvent::StreamStart => {
+                    continue
+                }
                 YamlEvent::DocumentEnd => return Some(document),
 
-                YamlEvent::Alias(alias) => match anchors.get(&alias) {
-                    Some(id) => Event::Alias(*id),
-                    None => {
+                YamlEvent::Alias(alias) => {
+                    if let Some(id) = anchors.get(&alias) {
+                        Event::Alias(*id)
+                    } else {
                         document.error = Some(
                             error::new(ErrorImpl::UnknownAnchor(mark))
                                 .shared(),
                         );
                         return Some(document);
                     }
-                },
+                }
+
                 YamlEvent::Scalar(mut scalar) => {
+                    // NEW: Duplicate anchor check
                     if let Some(anchor) = scalar.anchor.take() {
+                        if anchors.contains_key(&anchor) {
+                            document.error = Some(
+                                error::new(ErrorImpl::DuplicateAnchor(
+                                    mark.to_string(),
+                                ))
+                                .shared(),
+                            );
+                            // If you want to end parsing entirely, uncomment:
+                            self.parser = None;
```

Comment:
```suggestion
                            // If you want to end parsing entirely, uncomment:
                            self.parser = None;
```

Is this intentionally uncommented?

Reply:
@AaronDewes at the moment yes. Working on on https://github.com/sebastienrousseau/libyml/pull/7 before going back to https://github.com/sebastienrousseau/serde_yml/pull/41